src/Http/Middleware/HandleErrors.php

```diff
@@ -52,7 +52,7 @@ public function __invoke($error, Request $request, Response $response, callable
             $status = $errorCode;
         }
 
-        if ($this->debug && $errorCode !== 404) {
+        if ($this->debug && ! in_array($errorCode, [403, 404])) {
```

Comment:
I think one of the main points of debug mode is that Exceptions are exposed, with all debug information, and Whoops is the perfect way to do that. 404 is an exception (pun intended) to the rule because they almost always original from the same place – the route dispatcher. Users shouldn't have debug mode on unless they are developing or debugging an issue, in which case showing as much information as possible is necessary and pretty error pages are not.